rules-reviewed/hibernate/hibernate.windup.xml

```diff
@@ -30,6 +30,65 @@
                     </hint>
                 </iteration>
             </perform>
+        </rule>	
+	<!-- This rule may throw exception because some of the operations may not have it's condition part-->
+	<rule id="hibernate-02000">
+            <when>
+		<or>
+                <javaclass references="org.hibernate.criterion.Projections.count{*}" as="1">
+			<location>METHOD_CALL</location>
+		</javaclass>
+		<javaclass references="org.hibernate.criterion.Projections.rowCount{*}" as="2">
+			<location>METHOD_CALL</location>
+		</javaclass>
+		<javaclass references="org.hibernate.criterion.Projections.sum{*}" as="3">
+			<location>METHOD_CALL</location>
+		</javaclass>
+		</or>
+            </when>
+            <perform>
+                <iteration over="1">
+                    <hint title="@Type(type=text) is deprecated in Hibernate 4" effort="1">
+                        <message>
+                            The return types from the following projections has changed to Long in hibernate 4.
+                        </message>
+                        <link href="http://docbuilder.usersys.redhat.com/13710/#Migrate_Your_Hibernate_33x_Application_to_Hibernate_4x" title="Hibernate 4 projections return type change." />
+                    </hint>
+                </iteration>
+		<iteration over="2">
+                    <hint title="@Type(type=text) is deprecated in Hibernate 4" effort="1">
+                        <message>
+                            The return types from the following projections has changed to Long in hibernate 4.
+                        </message>
+                        <link href="http://docbuilder.usersys.redhat.com/13710/#Migrate_Your_Hibernate_33x_Application_to_Hibernate_4x" title="Hibernate 4 projections return type change." />
+                    </hint>
+                </iteration>
+		<iteration over="3">
+                    <hint title="@Type(type=text) is deprecated in Hibernate 4" effort="1">
+                        <message>
+                            The return types from the following projections has changed to Long in hibernate 4.
+                        </message>
+                        <link href="http://docbuilder.usersys.redhat.com/13710/#Migrate_Your_Hibernate_33x_Application_to_Hibernate_4x" title="Hibernate 4 projections return type change." />
+                    </hint>
+                </iteration>
+            </perform>
+        </rule>
+	<rule id="hibernate-03000">
+            <when>
+                <javaclass references="org.hibernate.cfg.AnnotationConfiguration{*}">
+			<location>CONSTRUCTOR_CALL</location>
+		</javaclass>
+            </when>
+            <perform>
+                <iteration>
+                    <hint title="The initial naming strategy was changed from DefaultNamingStrategy to EJB3NamingStrategy in Hibernate 4" effort="1">
+                        <message>
+                            This can result in naming mismatches.
```

Comment:
Please use more descriptive messages.

"Doing X with the hibernate API can result in naming mismatches between Y and Z. It is recommended to ABCD instead..."